install_deps.sh

```diff
@@ -46,6 +46,9 @@ function y_or_n() {
 
 if [ "${ID_LIKE}" = debian ] || [ "${ID}" = debian ]
 then
+	# Set nproc to either 1 or half the available cores
+	illixr_nproc=$(python3 -c 'import multiprocessing; print( max(multiprocessing.cpu_count() // 2, 1))')
```

Comment:
Is there an easy way to make this more intelligent / adaptive? This is only required for machines with low RAM, and possibly even then only for GTSAM, and we would be leaving a lot of performance on the table on machines with more RAM.

Reply:
I share your frustration, though I don't know of any examples which do something smarter here.